src/app/website/shared/testimonials/carousel.component.ts

```diff
@@ -0,0 +1,178 @@
+    }
+
+    // Prevent this user-triggered transition from occurring if there is already one in progress
+    if (nextSlide && nextSlide !== this.currentSlide) {
+      this.goNext(nextSlide, direction);
+    }
+  }
+
+  private goNext(slide: Slide, direction: Direction) {
+    if (this.destroyed) {
+      return;
+    }
+
+    slide.direction = direction;
+    slide.active = true;
+
+    if (this.currentSlide) {
+      this.currentSlide.direction = direction;
+      this.currentSlide.active = false;
+    }
+
+    this.currentSlide = slide;
+
+    // every time you change slides, reset the timer
+    this.restartTimer();
+  }
+
+  private getSlideByIndex(index: number) {
+    let len = this.slides.length;
+    for (let i = 0; i < len; ++i) {
+      if (this.slides[i].index === index) {
+        return this.slides[i];
+      }
+    }
+  }
+
+  private getCurrentIndex() {
+    return !this.currentSlide ? 0 : this.currentSlide.index;
+  }
+
+  private next() {
+    let newIndex = (this.getCurrentIndex() + 1) % this.slides.length;
+
+    if (newIndex === 0 && this.noWrap) {
+      this.pause();
+      return;
+    }
+
+    return this.select(this.getSlideByIndex(newIndex), Direction.NEXT);
+  }
+
+  private prev() {
+    let newIndex = this.getCurrentIndex() - 1 < 0 ? this.slides.length - 1 : this.getCurrentIndex() - 1;
+
+    if (this.noWrap && newIndex === this.slides.length - 1) {
+      this.pause();
+      return;
+    }
+
+    return this.select(this.getSlideByIndex(newIndex), Direction.PREV);
+  }
+
+  private restartTimer() {
+    this.resetTimer();
+    let interval = +this.interval;
+    if (!isNaN(interval) && interval > 0) {
+      this.currentInterval = setInterval(() => {
+        let nInterval = +this.interval;
+        if (this.isPlaying && !isNaN(this.interval) && nInterval > 0 && this.slides.length) {
+          this.next();
+        } else {
+          this.pause();
+        }
+      }, interval);
+    }
+  }
+
+  private resetTimer() {
+    if (this.currentInterval) {
+      clearInterval(this.currentInterval);
+      this.currentInterval = null;
+    }
+  }
+
+  public play() {
+    if (!this.isPlaying) {
+      this.isPlaying = true;
+      this.restartTimer();
+    }
+  }
+
+  public pause() {
+    if (!this.noPause) {
+      this.isPlaying = false;
+      this.resetTimer();
+    }
+  }
+
+  public addSlide(slide: Slide) {
+    slide.index = this.slides.length;
+    this.slides.push(slide);
+    if (this.slides.length === 1 || slide.active) {
+      this.select(this.slides[this.slides.length - 1]);
+      if (this.slides.length === 1) {
+        this.play();
+      }
+    } else {
+      slide.active = false;
+    }
+  }
+
+  public removeSlide(slide: Slide) {
+    this.slides.splice(slide.index, 1);
+
+    if (this.slides.length === 0) {
+      this.currentSlide = null;
+      return;
+    }
+
+    for (let i = 0; i < this.slides.length; i++) {
+      this.slides[i].index = i;
+    }
+  }
+}
+
```

Comment:
I think this is just copy paste. I am fine with it but as long as you understand the concepts. If you don't understand and if copy paste becomes an habit, I will have to question that approach. Please remember, you have to learn a lot of concepts and it's in your interest that you don't copy paste unless you understand how it is working